### PR TITLE
Refactor the util functions

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
The following util functions have been refactored by this pr.

- queryConcat
- arrayFlattenQuery

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/2013

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

**Comment by daneshk** [comment]
@kalaiyarasiganeshalingam shall we add another test case where the last param query doesn't contain any dynamic values

File: ballerina/utils.bal [hunk 1]
```diff
@@ -32,41 +30,29 @@ public isolated function queryConcat(ParameterizedQuery... queries) returns Para
 
 isolated function prepareParameterizedQuery(ParameterizedQuery[] queries) returns ParameterizedQuery {
     ParameterizedQuery newParameterizedQuery = ``;
-    string queryInString = "";
-    string nullValue = "";
     string[] strings = [];
     Value[] values = [];
-    int i = 0;
+    string previousString = "";
+    string firstStringValue = "";
```
Comment (daneshk) [inline ballerina/utils.bal:36]:
```suggestion
```

**Comment by kalaiyarasiganeshalingam** [comment]
> @kalaiyarasiganeshalingam shall we add another test case where the last param query doesn't contain any dynamic values

I added this test case for this. Are you expecting this?
https://github.com/ballerina-platform/module-ballerina-sql/pull/399/files#diff-bec1439bb904df4d16e12427eee7b2f1a0dc568bfefdbc6e804135d9cd8243e7R240

File: ballerina/utils.bal [hunk 2]
```diff
@@ -32,41 +30,29 @@ public isolated function queryConcat(ParameterizedQuery... queries) returns Para
...
         }
+        if (length >= 1) {
+            previousString = previousString + stringValues[stringValues.length() -1];
+        }
+        addValues(insertionValues, values);
     }
-    if queryInString != nullValue {
-        strings.push(queryInString);
-    }
+    strings.push(previousString);
```
[the remark this answers is not in the document]
Comment (daneshk) [reply ballerina/utils.bal:55]: shall we make `previousString` optional and check whether it has a string. like,

```
  if previousString is string {
      strings.push(previousString);
  }
```

**Review by daneshk — APPROVED** [review]

**Pull request merged**